pkg/generator/caches_test.go

```diff
@@ -166,14 +166,14 @@ func TestDeleteIngressInfoWhenDoesNotExist(t *testing.T) {
 	assert.DeepEqual(t, listenersBeforeDelete, listenersAfterDelete, protocmp.Transform())
 }
 
-func TestTLSListenerWithEnvCertsSecret(t *testing.T) {
+func TestExternalTLSListener(t *testing.T) {
```

Comment:
We don't want to support
- TestExternalTLSListenerWithEnvCertsSecrets 
- TestExternalTLSListenerWithInternalCertSecret
Local is only meant to use the config map config?

Reply:
I don't think I get your question.

The ExternalTLS Listener is for cluster-external domains and the cluster-local one is for the cluster-local domains. Both can have the SNI option (has already been there for the external one, this PR adds it for cluster-local) when a specific cert is there for a specific domain, and a general certificate for all other traffic (that is already there for both listeners).

Reply:
Yes but the tests were renamed and the source of the cert can be either env or some cm config. What do we support for internal and external cases?

Reply:
I felt the naming did not really reflect what they test. Both cases can have a default cert for all domains and specific ones for specified domains, I think we do cover all of them for both cases with:

```
func TestExternalTLSListener(t *testing.T) {

// sets the default cert
t.Setenv(envCertsSecretNamespace, "certns")
t.Setenv(envCertsSecretName, "secretname")

// Test with only default domains
t.Run("without SNI matches", func(t *testing.T) {}

// Test with one added specific domain
t.Run("with a single SNI match", func(t *testing.T) {}

// Test with two added specific domains
t.Run("with multiple SNI matches", func(t *testing.T) {}
}
```